File: SwrveSDK/SDK/Messaging/SwrveButtonThemeState.m
#import "SwrveButtonThemeState.h"

@implementation SwrveButtonThemeState

@synthesize fontColor;
@synthesize bgColor;
@synthesize borderColor;
@synthesize bgImage;

- (id)initWithDictionary:(NSDictionary *)themeState {
    if (self = [super init]) {
        if ([themeState objectForKey:@"font_color"]) {
            self.fontColor = [themeState objectForKey:@"font_color"];
        }
        if ([themeState objectForKey:@"bg_color"]) {
            self.bgColor = [themeState objectForKey:@"bg_color"];
        }
        if ([themeState objectForKey:@"border_color"]) {
            self.borderColor = [themeState objectForKey:@"border_color"];
        }
        if ([themeState objectForKey:@"bg_image"] && [themeState objectForKey:@"bg_image"] != [NSNull null]) {
            self.bgImage = [(NSDictionary *) [themeState objectForKey:@"bg_image"] objectForKey:@"value"];
        }
    }
    return self;
}

@end

File: public/SwrveSDKSwift/SwrveButtonThemeState.swift
import Foundation

@objc public class SwrveButtonThemeState: NSObject {

    @objc public var fontColor: String?
    @objc public var bgColor: String?
    @objc public var borderColor: String?
    @objc public var bgImage: String?

    @objc public init(dictionary themeState: [String: Any]) {
        if let fontColor = themeState["font_color"] as? String {
            self.fontColor = fontColor
        }
        if let bgColor = themeState["bg_color"] as? String {
            self.bgColor = bgColor
        }
        if let borderColor = themeState["border_color"] as? String {
            self.borderColor = borderColor
        }
        if let bgImageAny = themeState["bg_image"], !(bgImageAny is NSNull),
            let bgImageDict = bgImageAny as? [String: Any]
        {
            self.bgImage = bgImageDict["value"] as? String
        }
    }
}
